tests/publiccloud/azure_cli.pm

```diff
@@ -80,4 +100,55 @@ sub test_flags {
     return {fatal => 0, milestone => 0, always_rollback => 1};
 }
 
+sub virtual_network_test {
+    my ($rg,$loc,$mn,$ssh,$img) = @_;
+    assert_script_run("cd $root_dir");
+    assert_script_run('curl ' . data_url('publiccloud/azure_vn.sh') . ' -o azure_vn.sh');
+    assert_script_run('chmod +x azure_vn.sh');
+    my $start_cmd = $root_dir . '/azure_vn.sh $rg, $loc, $mn, $ssh, $img start ' . $self->instance_log_args();
+    assert_script_run($start_cmd);
+}
+
+sub run_cmd_test {
+    my ($rg,$loc,$mn,$ssh,$img) = @_;
+    assert_script_run("cd $root_dir");
+    assert_script_run('curl ' . data_url('publiccloud/azure_runcmd.sh') . ' -o azure_runcmd.sh');
+    assert_script_run('chmod +x azure_runcmd.sh');
+    my $start_cmd = $root_dir . '/azure_runcmd.sh $rg, $loc, $mn, $ssh, $img start ' . $self->instance_log_args();
+    assert_script_run($start_cmd);
+}
+
+sub vmss_test {
+    my ($rg,$loc,$mn,$ssh,$img) = @_;
+    assert_script_run("cd $root_dir");
+    assert_script_run('curl ' . data_url('publiccloud/azure_vmss.sh') . ' -o azure_vmss.sh');
+    assert_script_run('chmod +x azure_vmss.sh');
+    my $start_cmd = $root_dir . '/azure_vmss.sh $rg, $loc, $mn, $ssh, $img start ' . $self->instance_log_args();
+    assert_script_run($start_cmd);
+}
+
+sub create_vm {
+    my ($ar,$br,$ctr) = @_;
+
+    select_serial_terminal();
+    my $provider = $args->{my_provider};
+    my $instance = $provider->create_instance();
+    $instance->wait_for_guestregister();
+    registercloudguest($instance) if is_byos();
+
+    # call addons for pcm and phub
+    # register module-public-cloud and PackageHub
+    register_addons_in_pc($instance);
+
+    #Add Repos and install azure-cli
+    $instance->run_ssh_command(cmd => 'sudo zypper -n addrepo -fG ' . $ar, timeout => 600);
+    $instance->run_ssh_command(cmd => 'sudo zypper -n addrepo -fG ' . $br, timeout => 600);
+    $instance->run_ssh_command(cmd => 'sudo zypper -n addrepo -fG ' . $ctr, timeout => 600);
+    $instance->ssh_assert_script_run('sudo zypper ref; sudo zypper -n up', timeout => 300);
+    $instance->ssh_assert_script_run('sudo zypper install --allow-vendor-change --force azure-cli', timeout => 300);
+
+    record_info('azure cli installed');
+
+    sleep 90;    # wait for a bit for zypper to be available
```

Comment:
what you are waiting for ?

Reply:
Please use `utils::script_retry()`. We should not use `sleep()`.